SingularAdData.js

```diff
@@ -0,0 +1,101 @@
+export class SingularAdData {
+
+    constructor(adPlatform, currency, revenue) {
+        this._values = {
+            ad_platform: adPlatform,
+            ad_currency: currency,
+            ad_revenue: revenue,
+            is_admon_revenue: true,
+            is_revenue_event: true,
+            ad_mediation_platform: adPlatform
+        };
+    }
+
+    getAdDataValues() {
+        return this._values;
+    }
+
+    setNetworkName(networkName) {
```

Comment:
Instead of using `set` methods like we did in iOS, let's use the `with` like we did in Android